src/main/dataset_anonymization/cpp_anonymization.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) 2020 Anastasiia Birillo, Elena Lyulina
+
+import subprocess
+from typing import Optional
+
+import pandas as pd
+
+from src.main.task_scoring.task_scoring import unpack_tests_results, TESTS_RESULTS
+from src.main.util.consts import LANGUAGE, FILE_SYSTEM_ITEM, EXTENSION, ISO_ENCODING, CODE_TRACKER_COLUMN, TEST_RESULT, \
+    TASK
+from src.main.util.file_util import remove_slash, get_parent_folder, get_all_file_system_items, task_item_condition, \
+    get_name_from_path, extension_file_condition, get_content_from_file, create_file, remove_directory
+
+
+def __anonymize_code_fragment(code_fragment: str, local_gorshochek_path: str) -> Optional[str]:
+    """
+    gorshochek works only with folders with cpp files. We create a folder with the code fragment and run gorshochek.
+
+    Note: the default scripts/run.sh file in the gorshochek repository requires sudo access for docker running.
+    We remove it for the anonymization process to avoid running sudo processes from an external process.
+
+    After getting the result we delete the crested folders.
+    """
+    print(f'Start handling the code fragment:\n{code_fragment}\n\n')
+    input_name = 'input'
+    data_name = 'data'
+    out_name = f'{data_name}_out'
+    local_gorshochek_path = remove_slash(local_gorshochek_path)
+    input_dir = f'{local_gorshochek_path}/{data_name}'
+    create_file(code_fragment, f'{input_dir}/{input_name}.cpp')
+    output_dir = f'{local_gorshochek_path}/{out_name}'
+    p = subprocess.Popen(['sh', f'./scripts/run.sh', data_name, out_name], cwd=local_gorshochek_path)
+    p.wait()
+    print('Try to get the result')
+    anonymized_code = get_content_from_file(f'{output_dir}/{input_name}/transformation_1.cpp')
+    remove_directory(input_dir)
+    remove_directory(output_dir)
```

Comment:
Shouldn't you only remove the files, not directories, to not to recreate them again?